apps/angular/src/core/services/anime.service.ts

```diff
@@ -1,30 +1,116 @@
-import { inject, Injectable } from '@angular/core';
+import { inject, Injectable, OnDestroy } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Anime } from '@js-camp/core/models/anime';
 import { AnimeDto } from '@js-camp/core/dtos/anime.dto';
-import { map, Observable } from 'rxjs';
+import { BehaviorSubject, map, Observable, Subject, switchMap, take, takeUntil, tap } from 'rxjs';
 import { AnimeMapper } from '@js-camp/core/mappers/anime.mapper';
 import { PaginationDto } from '@js-camp/core/dtos/pagination.dto';
+import { PaginationMapper } from '@js-camp/core/mappers/pagination.mapper';
+import { Pagination } from '@js-camp/core/models/pagination';
+import { AnimeParams } from '@js-camp/core/models/based-params';
+import { Sort } from '@angular/material/sort';
+import { MatSelectChange } from '@angular/material/select';
 
 import { ApiUrlService } from './api-url.service';
+import { QueryParamsService } from './query-params.service';
 
 /**
  * Anime service to interact with the API.
  */
 @Injectable({
 	providedIn: 'root',
 })
-export class AnimeService {
+export class AnimeService implements OnDestroy {
```

Comment:
```suggestion
export class AnimeService {
```
The service has no life hooks